File: API/Entities/BOMEntry.cs
namespace API.Entities
{
    public class BOMEntry
    {
        public int PartId { get; set; }
        public int BOMId { get; set; }
        public Part Part { get; set; }

        public float Quantity { get; set; }
        public string ComponentLocation { get; set; }

    }
}

File: client/src/app/_models/bomEntry.ts
import { Part } from "./part";

export interface BOMEntry {
    partId: number;
    bomId: number;
    part: Part;
    quantity: number;
    componentLocation: string;
}

export interface NewBOMEntry {
    partId: number;
    quantity: number;
    componentLocation: string;
}